### PR TITLE
Completed assignment

### PR DESCRIPTION
**Pull request by davelowqx**
Missing documentation.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 # Write your flask code here
 
 app = Flask(__name__)
 
+@app.route("/")
+def index():
+    return "<p>Welcome</p>"
+
 app.register_blueprint(profiles_api, url_prefix="/profiles")
 app.register_blueprint(auth_api, url_prefix="/auth")
```
Comment (nobelang78) [inline main.py:15]: There no code to actually start the server. Look into app.run

File: main.py [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
-from db import db
 
 
 # Write your flask code here
 
 app = Flask(__name__)
 
+@app.route("/")
+def index():
+    return "<p>Welcome</p>"
```
Comment (nobelang78) [inline main.py:12]: We are not returning HTML, this is a API, only return string/json/dicts

File: Profiles/ProfilesAPI.py [hunk 3]
```diff
@@ -1,7 +1,70 @@
...
+@profiles_api.route("/<name>", methods=["GET"])
+def read(name):
+	row = db_findkey(name)
+	if row:
+		return jsonify({"error": None, "data" : row}), 200
+	else: 
+		return jsonify({"error": "not found", "data": None}), 404 
+
+# creates new profile
+@profiles_api.route("/<name>", methods=["POST"])
```
Comment (nobelang78) [inline Profiles/ProfilesAPI.py:45]: Should be posting to "/"

File: Profiles/ProfilesAPI.py [hunk 4]
```diff
@@ -1,7 +1,70 @@
...
+		
+
 profiles_api = Blueprint("profiles", __name__)
+
+# gets data of matching profile 
+@profiles_api.route("/<name>", methods=["GET"])
+def read(name):
+	row = db_findkey(name)
+	if row:
+		return jsonify({"error": None, "data" : row}), 200
```
Comment (nobelang78) [inline Profiles/ProfilesAPI.py:40]: This is misleading, seems like a undefined error, if success just return a success message

File: Auth/AuthAPI.py [hunk 5]
```diff
@@ -1,7 +1,60 @@
...
+	if db_findkey(new_data["username"]):
+			raise ValueError("Profile already exists") 
+	data.append(new_data)
+	json.dump(data, open("./users.json", "w"))
+
 auth_api = Blueprint("auth", __name__)
+
+@auth_api.route("/login", methods=["POST"])
+def login():
+	token = request.args.get("token")
```
Comment (nobelang78) [inline Auth/AuthAPI.py:35]: Requirement is that will login via username + password (via form data) or token (via url arguments)/ First part not handled

**Review by codecodecoder78 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**